eclair-core/src/main/scala/fr/acinq/eclair/payment/send/MultiPartPaymentLifecycle.scala

```diff
@@ -85,7 +93,11 @@ class MultiPartPaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig,
         goto(PAYMENT_ABORTED) using PaymentAborted(d.sender, d.request, d.failures :+ LocalFailure(new RuntimeException("balance is too low")), d.pending.keySet)
       } else {
         val pending = setFees(d.request.routeParams, payments, payments.size + d.pending.size)
-        pending.foreach { case (childId, payment) => spawnChildPaymentFsm(childId) ! payment }
+        Kamon.runWithContextEntry(parentPaymentIdKey, cfg.parentId) {
```

Comment:
I find it surprising that this wraps the `pending.foreach` instead of being inside it (creating one new span for each child payment). Shouldn't the `Kamon.runWithContextEntry` be inside the `pending.foreach` and wraps only the `spawnChildPaymentFsm`?

Reply:
This doesn't create a span, it just sets the context, so that spans created by the children actors occur within that same context.